aviary/mission/gasp_based/ode/time_integration_base_classes.py

```diff
@@ -7,23 +7,53 @@
 
 from aviary.mission.gasp_based.ode.params import ParamPort
 from aviary.variable_info.enums import Verbosity
+from aviary.variable_info.variable_meta_data import _MetaData
 
 
 def add_SGM_required_inputs(group: om.Group, inputs_to_add: dict):
-    blank_execcomp = om.ExecComp()
+    blank_component = om.ExplicitComponent()
     for input, details in inputs_to_add.items():
-        blank_execcomp.add_input(input, **details)
+        blank_component.add_input(input, **details)
     group.add_subsystem('SGM_required_inputs',
-                        blank_execcomp,
+                        blank_component,
                         promotes_inputs=list(inputs_to_add.keys()),
                         )
 
 
-# Subproblem used as a basis for forward in time integration phases.
+def add_SGM_required_outputs(group: om.Group, outputs_to_add: dict):
+    iv_comp = om.IndepVarComp()
+    for output, details in outputs_to_add.items():
+        iv_comp.add_output(output, **details)
+    group.add_subsystem('SGM_required_outputs',
```

Comment:
Similar question for this source indepvarcomp.

Reply:
same as above, but for the state rates that are not directly calculated by the EOM